app/models/miq_worker/systemd_common.rb

```diff
@@ -116,18 +116,24 @@ def unit_config_file_path
     end
 
     def unit_config_file
-      # Override this in a sub-class if the specific instance needs
-      # any additional config
       <<~UNIT_CONFIG_FILE
         [Service]
-        MemoryHigh=#{worker_settings[:memory_threshold].bytes}
-        TimeoutStartSec=#{worker_settings[:starting_timeout]}
-        TimeoutStopSec=#{worker_settings[:stopping_timeout]}
-        WatchdogSec=#{worker_settings[:heartbeat_timeout]}
+        #{unit_config_hash.compact.map { |key, value| "#{key}=#{value}" }.join("\n")}
         #{unit_environment_variables.map { |env_var| "Environment=#{env_var}" }.join("\n")}
       UNIT_CONFIG_FILE
     end
 
+    # Override this in a sub-class if the specific instance needs
+    # any additional config
+    def unit_config_hash
+      {
+        "MemoryHigh"      => worker_settings[:memory_threshold]&.bytes,
+        "TimeoutStartSec" => worker_settings[:starting_timeout],
+        "TimeoutStopSec"  => worker_settings[:stopping_timeout],
+        "WatchdogSec"     => worker_settings[:heartbeat_timeout]
+      }
+    end
```

Comment:
Should this follow the pattern of the `unit_environment_variables` and return an array?

Maybe we can use an associative array or change both to be hashes so the translation from this structure to the unit file can be consistent.

Reply:
Hm possibly, these unit config values are distinctly different from the unit environment variables though.  The possible values here are a predefined set where the environment variables can be arbitrary

We could make `unit_config_*` return an array instead like `["MemoryHigh=#{worker_settings[:memory_threshold]&.bytes}"]` but that would make it more difficult for a subclass to e.g. `super.merge("MemoryHigh" => "something else")`

We could make `unit_environment_variables` return a hash like `{"DATABASE_HOSTNAME" => "localhost"}` instead of `["DATABASE_HOSTNAME=localhost"]` but we would still have to do the `"Environment=#{key}=#{var}"` part because every environment variable has to start with `"Environment=` so we couldn't use a hash for the entire thing.

I'd really like to keep the unit config a hash for simpler `super.merge()` behavior, so I think the only way to make these two more consistent would be for `unit_environment_variables` to return  `{"DATABASE_HOSTNAME" => "localhost"}` and then do
```
#{unit_config_hash.compact.map { |key, value| "#{key}=#{value}" }.join("\n")}
#{unit_environment_variables.map { |key, value| "Environment=#{key}=#{value}" }.join("\n")}
```

Let me know if you want to go that route

Reply:
Yeah, I think hashes would be my preference over associative arrays.  The hashes feel like the simplest way to hold the data and it can be separated from the mechanism that converts the hashes to the appropriate file format.  I think it's far easier to superclass and override these methods and know they need to provide key / values in the hash and the parent class will deal with writing the format correctly.